Dominik kopaczka branch

dkopaczka opened the pull request:
This is working version of nastran export. This backbone of nastran export is designed to easy 
expand all formats of nastran(LONG, SHORT, FREE).

dkopaczka committed:
Added Project Backbone

dkopaczka committed:
Working version of the project (Nastran Export)

dkopaczka committed:
Update version of Nastran Export

dkopaczka committed:
Add paths, CMakeList and comments(no changes in code)

dkopaczka commented:
On this branch is also Karol Kiełbasa code. We developed the code together. On begining file is comment about who created a file.

dkopaczka committed:
Clean code

dkopaczka committed:
Try to solve branch conflicts

dkopaczka committed:
Merge branch 'master' into DominikKopaczkaBranch

kamich commented on Model/Common/CommonMeshRepresentation.h:
```diff
@@ -1,53 +1,109 @@
-#pragma #ifndef MESHIS_COMMON_MESH_REPRESENTATION_H_
```
#pragma and #ifndef after doesn't make any sense to me. Is this correct?

kamich commented on Model/Common/CommonMeshRepresentation.h:
```diff
@@ -1,53 +1,109 @@
+
+
+#endif /* end of include guard: MESHIS_COMMON_MESH_REPRESENTATION_H_ */
+
```
After this line, the file looks duplicated. Was that your intention?

kamich commented on Model/IO/Export/MeshExporter.h:
```diff
@@ -1,27 +1,57 @@
+}//! end of namespace MeshIS
+
+#endif /* end of include guard: MESHIS_MESH_EXPORTER_H_ */
+=======
```
The "=======" symbol is probably an artifact from conflicted merge resolution. It should be removed, as well as one of parts above or below it.

kamich commented on Model/IO/Export/Nastran/BulkData/BulkData.h:
```diff
@@ -0,0 +1,40 @@
+
+using std::string;
+
+#ifndef CPPWINLIN_BULKDATA_H
```
The so-called "header guard" should be above any meaningfull statments in file.

kamich commented on Model/IO/Export/Nastran/BulkData/BulkData.h:
```diff
@@ -0,0 +1,40 @@
+					public:
+						BulkData();
+						~BulkData();
+					public: string virtual get_GRID_line_from_vertex(MeshIS::Model::Common::Vertex vertex, int vertexID) = 0;
```
What is a rationale to make this functions virtual?
And if they are made virtual, what other changes should be made to the class?

kamich commented on Model/IO/Export/Nastran/BulkData/LongBulkData.cpp:
```diff
@@ -0,0 +1,31 @@
+
+using namespace MeshIS::Model::Export::Nastran::BData;
+
+LongBulkData::LongBulkData()
```
If you are not creating any special constructor/destructor etc. just don't define them as empty. Consider using =default option to avoid typing empty code.

kamich commented on Model/IO/Export/Nastran/BulkData/LongBulkData.cpp:
```diff
@@ -0,0 +1,31 @@
+
+string LongBulkData::get_GRID_line_from_vertex(MeshIS::Model::Common::Vertex vertex, int vertexID)
+{
+	return string();
```
Perhaps those functions are not yet finished?

kamich commented on Model/IO/Export/Nastran/BulkData/LongBulkData.h:
```diff
@@ -0,0 +1,36 @@
+						LongBulkData();
+						~LongBulkData();
+
+					public: string virtual get_GRID_line_from_vertex(MeshIS::Model::Common::Vertex vertex, int vertexID) override;
```
It is very iInconvinient for me to read very long namespace nested names. Consider using some "using namespace::name" or typedefs.

kamich commented on Model/IO/Export/Nastran/BulkData/ShortBulkData.cpp:
```diff
@@ -0,0 +1,95 @@
+}
+
+
+ShortBulkData::~ShortBulkData()
```
Shouldn't be destructor virtual?

kamich commented on Model/IO/Export/Nastran/BulkData/ShortBulkData.h:
```diff
@@ -0,0 +1,83 @@
+				{
+					class ShortBulkData :public BulkData
+					{
+					#pragma region Constructors
```
#pragma region is MSVC specyfic and not-portable. AFAIK it is not a part of standard.

kamich commented on Model/IO/Export/Nastran/NastranExporter/NastranExporter.cpp:
```diff
@@ -0,0 +1,130 @@
+	os.open(p.c_str(), std::ios_base::out);
+	if (os.good())
+	{
+		create_bulk_data(mesh_data);
```
Any error handling?

kamich commented on Model/IO/Export/Nastran/NastranExporter/NastranExporter.h:
```diff
@@ -0,0 +1,93 @@
+		{
+			namespace Nastran
+			{
+				class NastranExporter : public MeshIS::Model::Export::MeshExporter {
```
Just comment: the class is ok, but source code looks quite ugly to me. It could be much simpler and clearer.

kamich commented on Model/IO/Import/MeshImporter.h:
```diff
@@ -1,27 +1,57 @@
+}//! end of namespace MeshIS
+
+#endif /* end of include guard: MESHIS_MESH_IMPORTER_H_ */
+=======
```
Not concluded merge here. Have you tried to compile this?

kamich reviewed (changes requested):
General code organisation is perfectly fine, there are however some errors and misuses. Good usage of modern C++ inside functions, but classes could be written better.

dkopaczka committed:
Correcting errors and misuses.

+Improved class structure.
+All methods have a maximum 5 lines of code(except for one switch/case
block in NastranExporter).
+Add comments to facilitate code analysis(The rest says code).
+Two files confirming the operation of the program.

dkopaczka committed:
Merge remote-tracking branch 'origin/DominikKopaczkaBranch' into DominikKopaczkaBranch

# Conflicts:
#	Model/Common/CommonMeshRepresentation.h
#	Model/IO/Export/MeshExporter.h
#	Model/IO/IO_Exceptions.h
#	Model/IO/Import/MeshImporter.h

dkopaczka committed:
Merge branch 'master' into DominikKopaczkaBranch

dkopaczka committed:
Changing the structure of classes

dkopaczka committed:
Merge branch 'master' into DominikKopaczkaBranch
